socceraction/spadl/opta.py

```diff
@@ -387,8 +387,11 @@ def players(self, game_id: int) -> DataFrame[OptaPlayerSchema]:
                 ids = _extract_ids_from_path(ffp, feed_pattern)
                 parser = self.parsers[feed](ffp, **ids)
                 _deepupdate(data, parser.extract_players())
-        df_players = pd.DataFrame(list(data.values()))
+        df_players = pd.DataFrame.from_dict(data, 'index')
+        df_players = df_players.reset_index()
+        df_players = df_players.rename(columns={'index': 'player_id'})
```

Comment:
original:

```player_name       object
team_id           object
position_code      int64
minutes_played     int64
dtype: object```

my version
```
player_id         object
player_name       object
team_id           object
position_code      int64
minutes_played     int64
game_id           object```

I can duplicate player_id to dict values (lines 999-1000 in opta.py) to avoid this changes.

Reply:
You should duplicate `player_id`. That is also how it is done in all other parsers.

The index is used to group information from different files. It does not necessarily have to be the player id. For example, if a data stream of a particular provider does not contain unique player ids, it could also be the player's name.